File: typescript/basic_grammar/operation/assignment_operator/disjunction.ts
export const disjunctionAssignmentOperator = () => {
  let a = 50
  a |= 21
  return a
}

export const decimalDisjunctionAssignmentOperator = () => {
  let a = 50
  a |= 21.9
  return a
}

export const negativeDisjunctionAssignmentOperator = () => {
  let a = 50
  a |= -21
  return a
}

console.log(disjunctionAssignmentOperator())
console.log(decimalDisjunctionAssignmentOperator())
console.log(negativeDisjunctionAssignmentOperator())

File: python/basic_grammar/operation/assignment_operator/disjunction.py
def disjunction_assignment_operator():
    a = 50
    a |= 21
    return a

# 以下のコードは実行できない
# def decimal_disjunction_assignment_operator():
#     a = 50
#     a |= 21.9
#     return a


def negative_disjunction_assignment_operator():
    a = 50
    a |= -21
    return a

print(disjunction_assignment_operator())
# print(decimal_disjunction_assignment_operator())
print(negative_disjunction_assignment_operator())